homeassistant/components/tplink/config_flow.py

```diff
@@ -410,9 +410,17 @@ async def _async_try_discover_and_update(
             self._discovered_device = await Discover.discover_single(
                 host, credentials=credentials
             )
-        except TimeoutError:
-            # Try connect() to legacy devices if discovery fails
-            self._discovered_device = await Device.connect(config=DeviceConfig(host))
+        except TimeoutError as ex:
+            # Try connect() to legacy devices if discovery fails. This is a
+            # fallback mechanism for legacy that can handle connections without
+            # discovery info but if it fails raise the original error which is
+            # applicable for newer devices.
+            try:
+                self._discovered_device = await Device.connect(
+                    config=DeviceConfig(host)
+                )
+            except:  # noqa: E722
```

Comment:
```suggestion
            except Exception:
```

bare except will catch `BaseException` which will swallow `SystemExit`, `KeyboardInterrupt`, `asyncio.CancelledError`

Reply:
Ai! not sure why I did that :flushed: Now fixed.